File: PIAO_FE/src/app/shared/models/classes/piao-dto.ts
import { CampiTecniciDTO } from './campi-tecnici-dto';
import { StakeHolderDTO } from './stakeholder-dto';

export class PIAODTO extends CampiTecniciDTO {
  id?: number;
  codPAFK?: string;
  denominazione?: string;
  versione?: string;
  tipologia?: string;
  tipologiaOnline?: string;
  statoPiao?: string;
  aggiornamento?: boolean;
  idSezione1?: number;
  idSezione21?: number;
  idSezione22?: number;
  idSezione23?: number;
  idSezione31?: number;
  idSezione32?: number;
  idSezione331?: number;
  idSezione332?: number;
  idSezione4?: number;

  private _stakeHolders?: StakeHolderDTO[];

  get stakeHolders(): StakeHolderDTO[] | undefined {
    return this._stakeHolders;
  }

  set stakeHolders(value: StakeHolderDTO[] | undefined) {
    this._stakeHolders = value?.sort((a, b) => (a.id ?? 0) - (b.id ?? 0));
  }
}

File: Common/Entity/src/main/java/it/ey/dto/PiaoDTO.java
package it.ey.dto;

import lombok.*;
import lombok.experimental.SuperBuilder;

import java.time.LocalDate;
import java.util.List;


@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@SuperBuilder(toBuilder = true)
public class PiaoDTO extends  CampiTecniciDTO {
    private Long id;
    private String codPAFK;
    private String denominazione;
    private String versione;
    private String tipologia;
    private String tipologiaOnline;
//    private String statoPiao;
    private List<StakeHolderDTO> stakeHolders;
    private Long idSezione1;
    private Long idSezione21;
    private Long idSezione22;
    private Long idSezione23;
    private Long idSezione31;
    private Long idSezione32;
    private Long idSezione331;
    private Long idSezione332;
    private Long idSezione4;
    private LocalDate dataApprovazione;
    private String url;


}
